### PR TITLE
Allow user of run_doxygen script to specify path to doxygen

### PR DESCRIPTION
**Pull request by thk123**
Modify the `run_doxygen.sh` shell script to optionally take the path to doxygen as an argument. 

This change made it a little easier to test the broken doxygen problem so I'm suggesting it as an improvement.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [x] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [na] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [na] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [na] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [x] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
:warning:
This PR failed Diffblue compatibility checks (cbmc commit: a17503376885afc456478e4b6e65c364d7b94fb6).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/124779570
Status will be re-evaluated on next push.
Common spurious failures include: the cbmc commit has disappeared in the mean time (e.g. in a force-push); the author is not in the list of contributors (e.g. first-time contributors); compatibility was already broken by an earlier merge.

**Comment by thk123** [comment]
Joel bot failure looks spurious.

File: scripts/run_doxygen.sh [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
+
 # Check doxygen version
 EXPECTED_VERSION="1.8.16"
-doxygen --version | grep ^$EXPECTED_VERSION > /dev/null
+$doxygen_executable --version | grep ^$EXPECTED_VERSION > /dev/null
 if [ $? -ne 0 ]
 then
   echo "WARNING: Using wrong version of doxygen.\
   The list of expected warnings is for version $EXPECTED_VERSION."
+  $doxygen_executable --version
```
Comment (owen-mc-diffblue) [inline scripts/run_doxygen.sh:12]: :pick: Ideally this would be in a separate commit

File: .travis.yml [hunk 2]
```diff
@@ -69,11 +69,10 @@ jobs:
           then
             scripts/build_doxygen.sh ${DOXYGEN_VERSION}
           fi
-        - export PATH="$PATH:${TRAVIS_BUILD_DIR}/doxygen/build/bin"
       script:
         - echo $PATH
-        - doxygen --version
-        - scripts/run_doxygen.sh
+        - ${TRAVIS_BUILD_DIR}/doxygen/build/bin/doxygen --version
+        - scripts/run_doxygen.sh ${TRAVIS_BUILD_DIR}/doxygen/build/bin/doxygen
```
Comment (owen-mc-diffblue) [inline .travis.yml:75]: As far as I understand (and my bash is pretty rusty), this commit isn't needed. Please explain why you want to make this change, preferably in the commit message.

Comment (chrisr-diffblue) [reply .travis.yml:75]: I assume this commit is needed to make sure that the newly built version of doxygen is picked up, rather than a pre-installed version (which would appear sooner on the PATH). If that's the case, then an alternative solution would be to just re-write the `export PATH` as : `export PATH="${TRAVIS_BUILD_DIR}/doxygen/build/bin:${PATH}"`

Comment (chrisr-diffblue) [reply .travis.yml:75]: I guess also that the added bonus here of explicitly specifying which doxygen binary is used by the run_doxygen.sh script, is that we make sure that script always works in that way for developers too.

File: scripts/run_doxygen.sh [hunk 3]
```diff
@@ -1,15 +1,19 @@
 #!/bin/bash
 
+doxygen_executable=${1:-doxygen}
```
Comment (chrisr-diffblue) [inline scripts/run_doxygen.sh:3]: Pondering this a bit more - Should we drop this magic "fallback to whatever doxygen is on the path" stuff, and instead just *always* require the script to have the doxygen binary specified? I'm completely open either way, just wondering what the "least surprises" route is.

**Review by chrisr-diffblue — APPROVED** [review]

File: .travis.yml (hunk 2 above)
Comment (thk123) [reply .travis.yml:75]: Yes it isn't needed - this all works now. The intent was to:
 - make it clearer which version is being used
 - help developer know how they can run a specific version of it using the script

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: a17503376885afc456478e4b6e65c364d7b94fb6).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/124779570

File: scripts/run_doxygen.sh (hunk 3 above)
Comment (owen-mc-diffblue) [reply scripts/run_doxygen.sh:3]: The fallback makes it easier to run locally. I don't think that it's surprising that the default is whatever `doxygen` resolves to.

**Review by peterschrammel — APPROVED** [review]

**Comment by NlightNFotis** [comment]
This will need a bit of re-work to resolve conflicts and get it to pass the current CI configuration.

Closing this now - if you think this was a mistake, please reopen the PR - it will need a rebase a on develop and a drop of the last commit (as we are no longer using travis)
